packages/valory/skills/abstract_round_abci/behaviour_utils.py

```diff
@@ -488,6 +489,26 @@ def __init__(self, **kwargs: Any):  # pylint: disable=super-init-not-called
         self._non_200_return_code_count: int = 0
         enforce(self.behaviour_id != "", "State id not set.")
 
+    @classmethod
+    def auto_behaviour_id(cls) -> str:
+        """
+        Get behaviour id automatically.
+
+        This method returns the auto generated id from the class name if the
+        class variable behaviour_id is not set on the child class.
+        Otherwise, it returns the class variable behaviour_id.
+        """
+        return (
+            cls.behaviour_id
+            if type(cls.behaviour_id) == str and cls.behaviour_id != ""
```

Comment:
On this class, the base class, the type of `cls.behaviour_id` is property. On a child class, if the class var is set a string, hence the check.